### PR TITLE
Ccc calibration update

### PR DESCRIPTION
**Pull request by local-minimum**
Bit restructure to allow better testing and return more information from the polynomial construction

### CODE REVIEW

File: scanomatic/data_processing/calibration.py [hunk 1]
```diff
@@ -683,11 +680,25 @@ def construct_polynomial(identifier, power):
...
+    calculated_sizes = calculate_sizes(data_store, poly)
+    slope, intercept, _, p_value, stderr = linregress(
+        data_store.target_value, calculated_sizes)
+
+    validation = validate_polynomial(slope, p_value, stderr)
 
     if validation is not CalibrationValidation.OK:
         return {
-            'validation': validation
+            'validation': validation.name,
```
Comment (gdetrez) [inline scanomatic/data_processing/calibration.py]: This is almost copied verbatim bellow (`ccc` is omitted here but that's probably not intended). Remove duplication.

File: scanomatic/data_processing/calibration.py [hunk 2]
```diff
@@ -701,10 +712,17 @@ def construct_polynomial(identifier, power):
     return {
         'ccc': identifier,
```
Comment (gdetrez) [inline scanomatic/data_processing/calibration.py]: This is an argument to the function, why return it?

File: scanomatic/data_processing/calibration.py [hunk 3]
```diff
@@ -701,10 +712,17 @@ def construct_polynomial(identifier, power):
...
         'polynomial_coefficients': poly_coeffs,
+        'polynomial_as_text': poly_as_text(poly_coeffs),
```
Comment (gdetrez) [inline scanomatic/data_processing/calibration.py]: Avoid returning derived data. Call `poly_as_text` where you actually use it. Or maybe write a `Polynomial` class with a `__str__` method?

File: scanomatic/data_processing/calibration.py [hunk 4]
```diff
@@ -701,10 +712,17 @@ def construct_polynomial(identifier, power):
...
         'polynomial_power': power,
```
Comment (gdetrez) [inline scanomatic/data_processing/calibration.py]: This is also an argument.

File: scanomatic/data_processing/test/test_calibration.py [hunk 5]
```diff
@@ -174,6 +177,10 @@ def test_construct_bad_polynomial(self, edit_bad_slope_ccc):
             response['validation'] is not
             calibration.CalibrationValidation.OK)
 
+        assert 'correlation' in response
+        assert 'measured_sizes' in response
+        assert 'calculated_sizes' in response
```
Comment (gdetrez) [inline scanomatic/data_processing/test/test_calibration.py]: Those are not very interesting tests. Can't you predict the actual values here?

File: scanomatic/data_processing/test/test_calibration.py [hunk 6]
```diff
@@ -185,6 +192,9 @@ def test_construct_good_polynomial(self, edit_ccc):
         response = calibration.construct_polynomial(
             identifier, power, access_token=token)
 
+        assert 'correlation' in response
+        assert response['correlation']['slope'] == pytest.approx(1, abs=0.02)
```
Comment (gdetrez) [inline scanomatic/data_processing/test/test_calibration.py]: What about the other parts of the correlation?

File: scanomatic/ui_server/calibration_api.py [hunk 7]
```diff
@@ -764,11 +764,18 @@ def construct_calibration(ccc_identifier, power):
...
+        extra_info += ' correlation: {}'.format(response['correlation'])
+
     return json_abort(
         400,
         reason="Construction refused. " +
-        "Validation of polynomial says: {}".format(
-            response["validation"].name
+        "Validation of polynomial says: {}{}".format(
+            response["validation"],
+            extra_info
```
Comment (gdetrez) [inline scanomatic/ui_server/calibration_api.py]: Test?

File: scanomatic/data_processing/calibration.py [hunk 8]
```diff
@@ -477,6 +477,7 @@ def transform_plate_slice(identifier, image_identifier, plate_id):
 
     if not grayscale_targets or not grayscale_values:
         _logger.error("The gray-scale values have not been setup")
+        return False
```
Comment (gdetrez) [inline scanomatic/data_processing/calibration.py]: Why not use an exception?

Comment (local-minimum) [reply scanomatic/data_processing/calibration.py]: Well none really except the method already used that pattern, but I can rewrite it so all cases raise and handle that raising in the endpoint if you want.

File: scanomatic/data_processing/calibration.py (hunk 1 above)
Comment (local-minimum) [reply scanomatic/data_processing/calibration.py]: fixed

File: scanomatic/data_processing/calibration.py (hunk 2 above)
Comment (local-minimum) [reply scanomatic/data_processing/calibration.py]: nope there's no point (fixed)

File: scanomatic/data_processing/calibration.py (hunk 4 above)
Comment (local-minimum) [reply scanomatic/data_processing/calibration.py]: yes (though causes ui fix) (fixed)

File: scanomatic/data_processing/test/test_calibration.py (hunk 5 above)
Comment (local-minimum) [reply scanomatic/data_processing/test/test_calibration.py]: fixed

File: scanomatic/data_processing/test/test_calibration.py (hunk 6 above)
Comment (local-minimum) [reply scanomatic/data_processing/test/test_calibration.py]: fixed

File: scanomatic/ui_server/calibration_api.py (hunk 7 above)
Comment (local-minimum) [reply scanomatic/ui_server/calibration_api.py]: not easy to make a nice test so I made something at least.

Comment (local-minimum) [reply scanomatic/ui_server/calibration_api.py]: added what seemed doable...

**Pull request merged**